tests/integration/fixtures/test_connect.py

```diff
@@ -1,3 +1,59 @@
-def test_databricks_connect(spark):
+import os
+from pytest import fixture
+from databricks.connect import DatabricksSession
+
+
+@fixture
+def serverless_env():
+    os.environ['DATABRICKS_SERVERLESS_COMPUTE_ID'] = "auto"
+    yield
+    os.environ.pop('DATABRICKS_SERVERLESS_COMPUTE_ID')
+
+
+@fixture
+def debug_env_bugfix(monkeypatch, debug_env):
+    # This is a workaround to set shared cluster
+    # TODO: Update secret vault for acceptance testing and remove the bugfix
+    monkeypatch.setitem(debug_env, "DATABRICKS_CLUSTER_ID", "1114-152544-29g1w07e")
+
+
+def test_databricks_connect(debug_env_bugfix, ws, spark):
     rows = spark.sql("SELECT 1").collect()
     assert rows[0][0] == 1
+
+    creator = get_cluster_creator(spark, ws)
+    assert creator  # non-serverless clusters must have assigned creator
+
+
+def test_databricks_connect_serverless(serverless_env, ws, spark):
+    rows = spark.sql("SELECT 1").collect()
+    assert rows[0][0] == 1
+
+    creator = get_cluster_creator(spark, ws)
+    assert not creator  # serverless clusters don't have assigned creator
+
+
+def test_databricks_connect_serverless_set_cluster_id(serverless_env, ws, request):
```

Comment:
What is this test testing exactly? 

I think I would wrap the `spark_serverless_new` in another fixture that exists within this test module only

Reply:
agree, it's more clean, I updated